### PR TITLE
Track CSAI ad media failures

### PR DESCRIPTION
**Pull request by daytime-em**

### CODE REVIEW

**Comment by daytime-em** [comment]
Screenshot 2024-04-02 at 1 37 46 PM

Tests are passing locally

File: MUXSDKImaListener/Classes/MuxImaListener.m [hunk 1]
```diff
@@ -95,6 +98,16 @@ - (MUXSDKAdEvent *_Nullable) dispatchEvent:(IMAAdEvent *)event {
             [_playerBinding dispatchAdEvent: playbackEvent];
             playbackEvent = [MUXSDKAdPlayingEvent new];
             break;
+        case kIMAAdEvent_LOG:
+            if (adData && adData[@"logData"]) {
+                NSDictionary *errorLog = (NSDictionary *)adData[@"logData"];
+                if (errorLog) {
+                    if (errorLog[@"errorCode"] || errorLog[@"errorMessage"] || errorLog[@"type"]) {
```
Comment (andrewjl-mux) [inline MUXSDKImaListener/Classes/MuxImaListener.m:105]: Are these keys coming from IMA? (If so wondering if we can add a doc link if there is one to help future us)

Comment (daytime-em) [reply MUXSDKImaListener/Classes/MuxImaListener.m:105]: They come from IMA, yes but there's no public constants for them. I got these values by failing requests with proxyman then dumping events the IMASDKEvents to the logs

Comment (daytime-em) [reply MUXSDKImaListener/Classes/MuxImaListener.m:105]: Screenshot 2024-04-04 at 3 29 17 PM

There's a log snippet of the keys above ⬆️

Comment (andrewjl-mux) [reply MUXSDKImaListener/Classes/MuxImaListener.m:105]: Got it!

**Review by andrewjl-mux — APPROVED** [review]
LGTM

**Pull request merged**